src/Promitor.Core.Scraping/AzureMonitorScraper.cs

```diff
@@ -107,10 +108,10 @@ protected virtual string DetermineMetricFilter(string metricName, TResourceDefin
         /// </summary>
         /// <param name="metricName">Name of the metric being queried</param>
         /// <param name="resourceDefinition">Contains the resource cast to the specific resource type.</param>
-        /// <param name="dimension">Provides information concerning the configured metric dimension.</param>
-        protected virtual string DetermineMetricDimension(string metricName, TResourceDefinition resourceDefinition, MetricDimension dimension)
+        /// <param name="dimensions">Provides information concerning the configured metric dimensions.</param>
+        protected virtual List<string> DetermineMetricDimensions(string metricName, TResourceDefinition resourceDefinition, List<MetricDimension> dimensions)
         {
-            return dimension?.Name;
+            return dimensions?.Select(dimension => dimension.Name).Where(dimensionName => !string.IsNullOrWhiteSpace(dimensionName)).ToList();
```

Comment:
We should also do a distinct

Reply:
Am I just not understanding this comment, or is there something missing at the end? 😄

Reply:
I'm not sure why I can't make a code suggestion but I'd replace `Where` with `Distinct` so that we don't have the same dimension name twice if somebody configures it that way.